cmd/restic/cmd_rewrite.go

```diff
@@ -230,6 +267,7 @@ func filterAndReplaceSnapshot(ctx context.Context, repo restic.Repository, sn *r
 	// Always set the original snapshot id as this essentially a new snapshot.
 	sn.Original = sn.ID()
 	sn.Tree = &filteredTree
+	sn.ProgramVersion = version
```

Comment:
Hmm, I'm not sure whether that is a good idea. Keeping the original program version is rather useful to know which restic version created a snapshot, which is the most likely point in time when something can go wrong.